udp-ballerina/listener.bal

```diff
@@ -0,0 +1,118 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/java;
+
+# This is used for creating UDP server endpoints. A UDP server endpoint is capable of responding to
+# remote callers. The `Listener` is responsible for initializing the endpoint using the provided configurations.
+public class Listener {
+
+    # Initializes the UDP listener based on the privovided configurations. 
+    # ```ballerina
+    #  listener Listener|error? server1 = new (8080);
+    # ```
+    # + localPort - The port number of the remote service
+    # + config - Configurations related to the `udp:Listener`
+    public isolated function init(int localPort, *ListenerConfig config) returns error? {
+        return externInitEndpoint(self,localPort, config);
+    }
+
+    # Binds a service to the `udp:Listener`.
+    # ```ballerina
+    # udp:error? result = udpListener.attach(helloService);
+    # ```
+    #
+    # + s - Type descriptor of the service
+    # + name - Name of the service
+    # + return - `()` or else a `udp:Error` upon failure to register
+    #             the listener
+    public isolated function attach(Service s, string[]|string? name = ()) returns error? {
+        return externRegister(self, s);
+    }
+
+    # Starts the registered service programmatically.
+    #
+    # + return - An `error` if an error occurred during the listener 
+    #            starting process
+    public isolated function 'start() returns error? {
+        var e = externStart(self);
+    }
+
+    # Stops the service listener gracefully. Already-accepted requests will be
+    # served before connection closure.
+    #
+    # + return - An `error` if an error occurred during the listener
+    #            stopping process
+    public isolated function gracefulStop() returns error? {
+        return externGracefulStop(self);
+    }
+
+    # Stops the service listener immediately. It is not implemented yet.
+    #
+    # + return - An `error` if an error occurred during the listener
+    #            stop process
+    public isolated function immediateStop() returns error? {
+        return ();
+    }
+
+    # Stops consuming messages and detaches the service from the `udp:Listener`.
+    # ```ballerina
+    # udp:error? result = udpListener.detach(helloService);
+    # ```
+    #
+    # + s - Type descriptor of the service
+    # + return - `()` or else a `udp:Error` upon failure to detach the service
+    public isolated function detach(Service s) returns error? {
```

Comment:
IMO, we need to remove the service here. Otherwise, even after calling detach, Listener will continue dispatching messages to the relevant resource right ?

Reply:
added detach